firefox-ios/Client/Frontend/AuthenticationManager/SensitiveViewController.swift

```diff
@@ -8,60 +8,71 @@ import Shared
 class SensitiveViewController: UIViewController {
     private var backgroundBlurView: UIVisualEffectView?
     private var isAuthenticated = false
-    private var sceneWillEnterForegroundNotificationObserver: NSObjectProtocol?
-    private var didEnterBackgroundNotificationObserver: NSObjectProtocol?
+    private var sceneWillEnterForegroundObserver: NSObjectProtocol?
+    private var didEnterBackgroundObserver: NSObjectProtocol?
+    private var willResignActiveObserver: NSObjectProtocol?
+    private var didBecomeActiveObserver: NSObjectProtocol?
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
-        sceneWillEnterForegroundNotificationObserver = NotificationCenter.default.addObserver(
-            forName: UIScene.willEnterForegroundNotification,
-            object: nil,
-            queue: .main
-        ) { [weak self] notification in
+        sceneWillEnterForegroundObserver = observe(UIScene.willEnterForegroundNotification) { [weak self] notification in
             guard let self,
                   let sensitiveWindowScene = self.view.window?.windowScene,
                   let notificationScene = notification.object as? UIWindowScene,
                   sensitiveWindowScene === notificationScene else { return }
             self.handleCheckAuthentication()
         }
 
-        didEnterBackgroundNotificationObserver = NotificationCenter.default.addObserver(
-            forName: UIApplication.didEnterBackgroundNotification,
-            object: nil,
-            queue: .main
-        ) { [weak self] notification in
+        didEnterBackgroundObserver = observe(UIApplication.didEnterBackgroundNotification) { [weak self] notification in
             guard let self = self else { return }
-
             self.isAuthenticated = false
             self.installBlurredOverlay()
         }
+
+        willResignActiveObserver = observe(UIApplication.willResignActiveNotification) { [weak self] notification in
+            guard let self = self else { return }
```

Comment:
Could use `guard let self else { return }` in this PR if you want, not sure if we prefer using one or the other in the codebase, there seems to be instances of both